sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -1735,6 +1735,7 @@ describe("Atom management - Authentication", function(): void {
     testCaseTitle: "Correlation Filter rule options with a single property",
     input: {
       filter: {
+        sqlExpression: "",
```

Comment:
@mohsin-mehmood Can you share some thoughts on why we had to do this?

Reply:
@ramya-rao-a `createEntity` function last argument is `ruleOptions?: Omit<RuleProperties, "name">` and `RuleProperties` interface has property `filter: SqlRuleFilter | CorrelationRuleFilter`. 

With `SqlRuleFilter` interface updated to have `sqlExpression` as required, test inputs of type  `CorrelationRuleFilter` are also expected to have `sqlExpression` property.

Reply:
@HarshaNalluru Is there something else we should be doing here?

Reply:
This is weird, I'm looking into it!

Reply:
There are multiple issues here.

- First thing, update the `createEntity` to take "CreateRuleOptions" instead of `RuleProperties`.
   `ruleOptions?: Omit<RuleProperties, "name">` -> `ruleOptions?: Omit<CreateRuleOptions, "name">`.**[TODO-1]**
- Remove the newly added `sqlExpression: ""` for the correlation rule test cases **[TODO-2]**

Upon doing the above, you'll see another problem which would be a TS complaint
   ```
     Type '{ correlationId: string; properties: { randomState: string; randomCountry?: undefined; randomCount?: undefined; 
     randomBool?: undefined; randomDate?: undefined; }; sqlExpression?: undefined; sqlParameters?: undefined; }' is not 
     assignable to type 'SqlRuleFilter'.          Types of property 'sqlExpression' are incompatible.
            Type 'undefined' is not assignable to type 'string'.
   ```

   Even though we don't set any of the properties to "undefined" in the correlation filter, TS tries to come up with a type that represents all the entries in the test cases array, that new type would have "undefined" value in the properties causing a mismatch with the CorrelationRuleFilter type. [Minimal repro here - for the TS limitation](https://www.typescriptlang.org/play?ts=4.0.2#code/JYOwLgpgTgZghgYwgAgJIgA4FczsrRFAbwChlkMB+ALmVPPIG0BrCAT1oGcwpQBzALq0QWALYAjaAG4yyAL4y5JEggD2IbsgAeyALzIAFKvEArWumy5w0eEgCUegHzJjJmSvWbQl85hx4bQkYBPWRGWSIKWkjWNgBGWgAmAGZ5eQAaCKi6ZFjE2mSAFjSlAXcAenLkADlVCihjABsIUWQ4MDbGxpJvHAA6GFUoAFFEAAsDIwddZy0pu3c1DQ7esETQ8PJIjGjc9gTkFJLMreyY9nzkIpKSMuVK5AAVNgwITgReDA61UQxGuFAnB6fjWAyGowQEymTm08ykQA)
- To overcome this issue, declare the test cases array with the type as shown at https://github.com/HarshaNalluru/azure-sdk-for-js/commit/b8fc66e69df8d82b7caf5a4da28f525669cf3de8 **[TODO-3]**

Please act as per the "TODO"s, you shouldn't see the problems upon fixing them.

Reply:
Thanks @HarshaNalluru!

@mohsin-mehmood If needed, lets pull out the change to `sqlExpression` from this PR and have a separate PR for it. That way, we can go ahead with all the other renames that are more trivial in this PR

Reply:
@HarshaNalluru Applied the changes as suggested